fdb-record-layer-lucene/src/main/java/org/apache/lucene/index/StandardDirectoryReaderOptimization.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * StandardDirectoryReaderOptimization.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.lucene.index;
```

Comment:
Does this need to be in this package (instead of a `com.apple.foundationdb.record.lucene`-prefixed package) to access `protected` methods or something along those lines?

Reply:
even worse, private.  Yes.

Reply:
I am going down the Lucene rabbit hole.